tests/python/gpu/test_optimizer.py

```diff
@@ -0,0 +1,93 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+import mxnet as mx
+
+import sys
+import os
+curr_path = os.path.dirname(os.path.abspath(os.path.expanduser(__file__)))
+sys.path.insert(0, os.path.join(curr_path, '../unittest'))
+from common import setup_module, with_seed
+
+# This script is testing the efficiency of LARS
+# We are training LeNet-5 at batch-size 8000 in 10 epochs above 98% accuracy
+# Which is not doable with simple SGD + momentum (from what have been tested so far)
+
+def lenet5():
+    """LeNet-5 Symbol"""
+    #pylint: disable=no-member
+    data = mx.sym.Variable('data')
+    conv1 = mx.sym.Convolution(data=data, kernel=(5, 5), num_filter=20)
+    tanh1 = mx.sym.Activation(data=conv1, act_type="tanh")
+    pool1 = mx.sym.Pooling(data=tanh1, pool_type="max",
+                           kernel=(2, 2), stride=(2, 2))
+    # second conv
+    conv2 = mx.sym.Convolution(data=pool1, kernel=(5, 5), num_filter=50)
+    tanh2 = mx.sym.Activation(data=conv2, act_type="tanh")
+    pool2 = mx.sym.Pooling(data=tanh2, pool_type="max",
+                           kernel=(2, 2), stride=(2, 2))
+    # first fullc
+    flatten = mx.sym.Flatten(data=pool2)
+    fc1 = mx.sym.FullyConnected(data=flatten, num_hidden=500)
+    tanh3 = mx.sym.Activation(data=fc1, act_type="tanh")
+    # second fullc
+    fc2 = mx.sym.FullyConnected(data=tanh3, num_hidden=10)
+    # loss
+    lenet = mx.sym.SoftmaxOutput(data=fc2, name='softmax')
+    #pylint: enable=no-member
+    return lenet
+
+@with_seed()
+def test_lars():
```

Comment:
how long does this test take? should it be put in nightly tests instead of unit tests? 

should we a test in `python/unittest/test_optimizer.py`?

Reply:
It's taking 10s with my V100

Reply:
So, I'm already testing the MXNet Ops in https://github.com/apache/incubator-mxnet/pull/16122/files#diff-4758fb9329d438de2836db2634a8f5f7R270-R422. which are the only non-python part of the optimizer, what my test is adding is to show that it's properly behaving: i.e allow you to train a network with a bigger batch. Making a fully python of it wouldn't help me testing that (the python could be wrong as well, let's say if I misunderstood the publication)

Reply:
@anirudhacharya Does swapping my training test to nightly test and just unit testing my MXNet Ops in `/tests/python/gpu/test_operator_gpu.py` enough ? If yes I believe I'm all done for this PR :)

Reply:
sorry, missed seeing this. i will take another look. And, yes it would be good to move time consuming tests to nightly tests. 

I understand that the python mock optimizer could be wrong if our understanding of the paper is incorrect. Let us try to catch them in the code review process.

Reply:
apart from the above mentioned test issue, everything else lgtm.